mods/parsoid.js

```diff
@@ -530,6 +551,45 @@ function cheapBodyInnerHTML(html) {
     }
 }
 
+/**
+ * Replaces sections in original content with sections provided in sectionsJson
+ */
+function replaceSections(original, sectionsJson) {
+    var sectionOffsets = original['data-parsoid'].body.sectionOffsets,
+        newBody = cheapBodyInnerHTML(original.html.body);
+    Object.keys(sectionsJson)
+    .sort(function(id1, id2) {
```

Comment:
We need to sort section in reverse order by section offset so that replaces don't clash with offset numbers.